admin-dev/themes/new-theme/js/pages/product/combination/bulk-form-handler.ts

```diff
@@ -0,0 +1,170 @@
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ConfirmModal from '@components/modal';
+import ProductMap from '@pages/product/product-map';
+import ComponentsMap from '@components/components-map';
+import ProductEvents from '@pages/product/product-event-map';
+import CombinationsService from '@pages/product/services/combinations-service';
+import {EventEmitter} from 'events';
+import DisablingToggler from '@components/form/disabling-toggler';
+
+const CombinationMap = ProductMap.combinations;
+const CombinationEvents = ProductEvents.combinations;
+
+export default class BulkFormHandler {
+  private combinationsService: CombinationsService;
+
+  private eventEmitter: EventEmitter;
+
+  private tabContainer: HTMLDivElement;
+
+  constructor() {
+    this.combinationsService = new CombinationsService();
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+    this.tabContainer = document.querySelector(CombinationMap.externalCombinationTab) as HTMLDivElement;
+    this.init();
+  }
+
+  private init(): void {
+    this.listenSelections();
+    this.eventEmitter.on(CombinationEvents.listRendered, () => this.toggleBulkActions());
+
+    const bulkFormBtn = document.querySelector(CombinationMap.bulkCombinationFormBtn) as HTMLButtonElement;
+    bulkFormBtn.addEventListener('click', () => this.showFormModal());
+  }
+
+  private showProgressModal(): ConfirmModal {
+    //@todo: Replace with new progress modal when introduced in #26004.
+    const modal = new ConfirmModal(
+      {
+        id: CombinationMap.bulkProgressModalId,
+        confirmMessage: '<div>Updating combinations: <p class="progress-increment"></p></div>',
+      },
+      () => null,
+    );
+
+    modal.show();
+
+    return modal;
+  }
+
+  private showFormModal(): ConfirmModal {
+    const template = document.querySelector(CombinationMap.bulkCombinationFormTemplate) as HTMLScriptElement;
+    const content = template.innerHTML;
+
+    const modal = new ConfirmModal(
+      {
+        id: CombinationMap.bulkCombinationModalId,
+        confirmMessage: content,
+      },
+      () => this.submitForm(),
+    );
+
+    modal.show();
+    const form = document.querySelector(CombinationMap.bulkCombinationForm) as HTMLFormElement;
+    const disablingToggles = form.querySelectorAll(ComponentsMap.disablingToggle.wrapper) as NodeListOf<HTMLDivElement>;
+
+    disablingToggles.forEach((element) => {
+      const {disablingToggleName} = element.dataset;
+      new DisablingToggler(
+        `${CombinationMap.bulkCombinationForm} [data-disabling-toggle-name="${disablingToggleName}"] input`,
+        '0',
+        `${CombinationMap.bulkCombinationForm} [data-toggled-by="${disablingToggleName}"]`,
```

Comment:
`data-disabling-toggle-name` should be turned into `data-disabling-toggle-target` and contain the name of the target input that needs to be disabled

Reply:
So you can do this:

```suggestion
        `${CombinationMap.bulkCombinationForm} [name="${disablingToggleTarget}"]`,
```

Reply:
Im pretty sure I've tried it (or something like this) but it didn't work (I don't remember why now). Will check one more time.

Reply:
![Screenshot from 2022-02-21 18-02-25](https://user-images.githubusercontent.com/31609858/154990293-52d7e7d1-5aa0-45a4-aff5-072ff3703f7a.png)
This is how single disabling toggle element looks like. It contains 2 inputs (as usual for switch). And input names are full (with the bulk_combination prefix, whilst in form listener I only have partial form names, thats why i added it into data attr).